### ISSUE TITLE
Fix error handling for custom build scripts

### ISSUE THREAD

**Issue by Artoria2e5** [AOSC-Dev/autobuild3#100]
Our current error handling for sourced scripts involve testing what is returned from the script itself. For predefined build scripts this works well because we have many return-on-error hooks, but things go wild with poorly-written (read: every single one) `autobuild/build` scripts.

We can either fix all the scripts, or at least partially switch to some "strict mode" handling by `set -ueE` and `trap something ERR`. (Hell, I don't even like how `.()` messes up scopes in the first place.) It is, however, still important to note that `set -e` does not properly handle lines with `&&` and/or `||`.

**Comment by Artoria2e5** [comment]
Yes, it is. See https://github.com/AOSC-Dev/autobuild3/blob/d146b34/proc/50-build_exec.sh#L6-L25.

**Comment by LionNatsu** [comment]
https://github.com/AOSC-Dev/autobuild3/blob/d146b34acfd90ad1b02579c47c6c4e306523c245/proc/50-build_exec.sh#L27

Look good, but why do `ABTYPE=self` never exit when errors occur?

For example, `autobuild/build`:
```shell
false
echo It should not be here
```

Is it because `set -e` cannot work properly in this scope?

**Comment by Artoria2e5** [comment]
We have a set -e ???????

**Comment by LionNatsu** [comment]
arch_loadfile_strict have.

**Comment by Artoria2e5** [comment]
It's probably the `||` thing that is breaking `set -e` here. Perhaps it should be rewritten into an explicit `if (($?)); then ...`
